include/phasar/PhasarLLVM/DataFlowSolver/Mono/Solver/MonoCache.h

```diff
@@ -0,0 +1,64 @@
+/******************************************************************************
+ * Copyright (c) 2020 Philipp Schubert.
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of LICENSE.txt.
+ *
+ * Contributors:
+ *     Ajay Subramanya Kudli Prasanna Kumar, Philipp Schubert and others
+ *****************************************************************************/
+
+#ifndef PHASAR_PHASARLLVM_MONO_SOLVER_MONOCACHE_H_
+#define PHASAR_PHASARLLVM_MONO_SOLVER_MONOCACHE_H_
+
+#include<iostream>
+#include<unordered_map>
+#include<vector>
+
+namespace psr {
+  template <class K, class V>
+class MonoCache {
+  // TODO
+public:
+        unordered_map<K, V> summaries;
+        vector<int> time; //The most recently added or recently used key is placed last
+        int c;
+
+        MonoCache(int capacity) {
+                c = capacity;
+        }
+
+void get(K key) {
+        unordered_map<K, V>::iterator iter = summaries.find(key);
+        if(iter == summaries.end())
+        return -1; //not found
+        else {
+                vector<int>::iterator it = std::find(time.begin(), time.end(), key);
+                time.erase(it); //The visited element is placed at the end
+                time.push_back(key);
+                return iter->second; //value
+        }
+
+}
+
+
+void put(K key, V value) {
+        if(summaries.count(key)) //key already exists in cache
+        {
+            summaries[key] = value; //key is set to the new value
+            vector<int>::iterator it = std::find(time.begin(), time.end(), key);
+            time.erase(it); //Update key usage time
+            time.push_back(key);
```

Comment:
I'm not sure what you are trying to do. Key is of arbitrary type `k`. But your time vector is a vector of ints.

Reply:
Changed the vector to type K in commit f67db78d18d8c4d1110699e239ef4e9d945f1789